NSubstitute.DbConnection/NSubstitute.DbConnection/DbConnnectionMockExtensions.cs

```diff
@@ -179,7 +187,12 @@ public IMockQueryResultBuilder ThenReturns<T>(params T[] results)
 
             public bool Matches(IDbCommand command)
             {
-                if (command.CommandText != CommandText)
+                if (Matcher != null)
+                {
+                    return Matcher.Invoke(command);
```

Comment:
I think you can just do Matcher(command) rather than Matcher.Invoke(command) - at least if it's a Func<T, bool> you definitely can!